### PR TITLE
Issue #SB-16572 feat:Renderer: Make default font size is 18px even though user is selected below 18px

### PR DESCRIPTION
**Pull request by JagadishPujari**
[JIRA LINK](https://project-sunbird.atlassian.net/browse/SB-16572)

### CODE REVIEW

File: player/public/coreplugins/org.ekstep.ecmlrenderer-1.0/renderer/plugin.js [hunk 1]
```diff
@@ -184,6 +184,7 @@ org.ekstep.contentrenderer.baseLauncher.extend({
         content.canvasId = dataObj.canvasId;
         EkstepRendererAPI.setRenderer(instance);
         Renderer.theme = new ThemePlugin(content);
+        qspatch.setDefaultFontSize(content);
```
Comment (vinukumar-vs) [inline player/public/coreplugins/org.ekstep.ecmlrenderer-1.0/renderer/plugin.js]: This should be just before 
line 202: qspatch.handleAssetUrl();

File: player/public/coreplugins/org.ekstep.ecmlrenderer-1.0/renderer/qspatch.js [hunk 2]
```diff
@@ -390,5 +390,70 @@ var qspatch = {
...
+    },
+    //Make default font size for question options
+    setDefaultFontSize: function(data){
+      if(data.stage['org.ekstep.questionset']){
+        try {
+          var questionSetData = data.stage['org.ekstep.questionset'];
+          _.each(questionSetData['org.ekstep.question'],function(item,key){
+              var questionData = JSON.parse(item.data.__cdata);
+              questionData.question.text = qspatch.changeFontSize(questionData.question);
+              item.data.__cdata = JSON.stringify(questionData);
```
Comment (vinukumar-vs) [inline player/public/coreplugins/org.ekstep.ecmlrenderer-1.0/renderer/qspatch.js]: Why do you need this line? You can update the question after changing the question & options font-size

File: player/public/coreplugins/org.ekstep.ecmlrenderer-1.0/renderer/qspatch.js [hunk 3]
```diff
@@ -390,5 +390,70 @@ var qspatch = {
...
+              item = qspatch.questionFontSizeChange(item);
+              questionSetData['org.ekstep.question'][key] = item;
+          });
+          // Renderer.theme = new ThemePlugin(questionSetData);
+        } catch (e) {
+            console.log(e);
+        }
+      }
+    },
+    questionFontSizeChange: function(item){
```
Comment (vinukumar-vs) [inline player/public/coreplugins/org.ekstep.ecmlrenderer-1.0/renderer/qspatch.js]: Change the function name 
"checkOptionsFontSize"

File: player/public/coreplugins/org.ekstep.ecmlrenderer-1.0/renderer/qspatch.js [hunk 4]
```diff
@@ -390,5 +390,70 @@ var qspatch = {
...
+              questionOptionsData.option.optionsLHS = qspatch.optionsTextFontChange(questionOptionsData.option.optionsLHS,item.type);
+              questionOptionsData.option.optionsRHS = qspatch.optionsTextFontChange(questionOptionsData.option.optionsRHS,item.type);
+              questionItem.data.__cdata = JSON.stringify(questionOptionsData);
+              break;
+        default: break;
+
+      }
+      return questionItem;
+    },
+    optionsTextFontChange: function(options,type){
```
Comment (vinukumar-vs) [inline player/public/coreplugins/org.ekstep.ecmlrenderer-1.0/renderer/qspatch.js]: Change the function name
"changeOptionsFontSize"

File: player/public/coreplugins/org.ekstep.ecmlrenderer-1.0/renderer/qspatch.js [hunk 5]
```diff
@@ -390,5 +390,70 @@ var qspatch = {
...
+        _.each(options,function(option,key){
+          optionsData[key].text = '<p style="font-size:1.28em">' + optionsData[key].text + '</p>';
+        });
+      }
+      return optionsData;
+    },
+    changeFontSize: function(data){
+        var element = $($.parseHTML(data.text));
+        console.log("Jags",$(element)[0])
+        if($(element) && $(element)[0] && $(element)[0].children && $(element)[0].children[0] && $(element)[0].children[0].style && $(element)[0].children[0].style.fontSize){
```
Comment (vinukumar-vs) [inline player/public/coreplugins/org.ekstep.ecmlrenderer-1.0/renderer/qspatch.js]: Data element[0] & element)[0].children[0] reference & use those objects to check. 
The code is messy here.

File: player/public/coreplugins/org.ekstep.ecmlrenderer-1.0/renderer/plugin.js (hunk 1 above)
Comment (JagadishPujari) [reply player/public/coreplugins/org.ekstep.ecmlrenderer-1.0/renderer/plugin.js]: Done

File: player/public/coreplugins/org.ekstep.ecmlrenderer-1.0/renderer/qspatch.js (hunk 2 above)
Comment (JagadishPujari) [reply player/public/coreplugins/org.ekstep.ecmlrenderer-1.0/renderer/qspatch.js]: Question object reference was lost so updating question and option text separately

File: player/public/coreplugins/org.ekstep.ecmlrenderer-1.0/renderer/qspatch.js (hunk 3 above)
Comment (JagadishPujari) [reply player/public/coreplugins/org.ekstep.ecmlrenderer-1.0/renderer/qspatch.js]: Done

File: player/public/coreplugins/org.ekstep.ecmlrenderer-1.0/renderer/qspatch.js (hunk 4 above)
Comment (JagadishPujari) [reply player/public/coreplugins/org.ekstep.ecmlrenderer-1.0/renderer/qspatch.js]: Done

File: player/public/coreplugins/org.ekstep.ecmlrenderer-1.0/renderer/qspatch.js (hunk 5 above)
Comment (JagadishPujari) [reply player/public/coreplugins/org.ekstep.ecmlrenderer-1.0/renderer/qspatch.js]: Done

File: player/public/coreplugins/org.ekstep.ecmlrenderer-1.0/renderer/qspatch.js [hunk 6]
```diff
@@ -390,5 +390,71 @@ var qspatch = {
...
+    },
+    changeOptionsFontSize: function(options,type){
+      var optionsData = options;
+      if(type == 'mcq'){
+        _.each(options,function(option,key){
+          optionsData[key].text = qspatch.changeFontSize(option);
+        });
+      }else if(type == 'mtf'){
+        _.each(options,function(option,key){
+          optionsData[key].text = '<p style="font-size:1.28em">' + optionsData[key].text + '</p>';
```
Comment (vinukumar-vs) [inline player/public/coreplugins/org.ekstep.ecmlrenderer-1.0/renderer/qspatch.js]: Still it is hardcoded here
@rahulshukla Review the PR

File: player/public/coreplugins/org.ekstep.ecmlrenderer-1.0/renderer/qspatch.js [hunk 7]
```diff
@@ -390,5 +390,71 @@ var qspatch = {
...
+      }
+      return optionsData;
+    },
+    changeFontSize: function(data){
+      var index = data.text.indexOf("<p><span");
+
+      if(index == 0){
+          var element = $($.parseHTML(data.text));
+          var size = $(element)[0].children[0].style.fontSize;
+          if(parseFloat(size) < parseFloat(globalConfig.questionMinFontSize)){
```
Comment (vinukumar-vs) [inline player/public/coreplugins/org.ekstep.ecmlrenderer-1.0/renderer/qspatch.js]: Use <=

File: player/public/coreplugins/org.ekstep.ecmlrenderer-1.0/renderer/qspatch.js [hunk 8]
```diff
@@ -390,5 +390,71 @@ var qspatch = {
...
+        _.each(options,function(option,key){
+          optionsData[key].text = '<p style="font-size:1.28em">' + optionsData[key].text + '</p>';
+        });
+      }
+      return optionsData;
+    },
+    changeFontSize: function(data){
+      var index = data.text.indexOf("<p><span");
+
+      if(index == 0){
```
Comment (vinukumar-vs) [inline player/public/coreplugins/org.ekstep.ecmlrenderer-1.0/renderer/qspatch.js]: Why still we are using index. Get fontSize first, if it is undefined, then do 
else part

Comment (JagadishPujari) [reply player/public/coreplugins/org.ekstep.ecmlrenderer-1.0/renderer/qspatch.js]: What if <p> tag itself is not there? How we have to get children and elements?

**Pull request merged**